pkg/resid/resid.go

```diff
@@ -95,3 +95,15 @@ func (id ResId) GvknEquals(o ResId) bool {
 func (id ResId) Equals(o ResId) bool {
 	return id.Namespace == o.Namespace && id.GvknEquals(o)
 }
```

Comment:
this is tricky.

namespaces were grafted on, depend on context, and we need to work out something that makes sense for kubectl apply.

maybe...

```
func (id ResId) Equals(o ResId) bool {
  return id.IsNsEquals(o) && id.GvnEquals(o)
}
```
```
func (id ResId) IsNsEquals(o ResId) bool {
  return id.Namespace == o.Namespace ||
   (id.IsInDefaultNs() && o.IsInDefaultNs() ) ||
    (!is.IsNamespaceable() && !is.IsNamespaceable())
}
```
```
func (id ResId) IsInDefaultNs() bool {
  return id.IsNamespaceable() && (id.Namespace == "" || id.Namespace == "default")
}
```
```
func (id ResId) IsNamespaceable() bool {
// implement
// https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#not-all-objects-are-in-a-namespace
}
```

Reply:
@monopole The requested functions have been added and implemented